test/stat_tracker_test.rb

```diff
@@ -58,22 +58,22 @@ def test_it_can_count_game_by_season
       "20132014" => 6
     }
 
-    assert_equal expected, @stat_tracker.count_of_games_by_season
+    assert_equal expected, @stat_tracker.count_of_games_per_season
   end
 
   def test_it_can_calculate_average_goals_per_game
     assert_equal 3.91, @stat_tracker.average_goals_per_game
   end
 
-  def test_it_can_return_hash_of_average_goals_by_season
+  def test_it_can_return_hash_of_average_goals_per_season
     expected_hash = {
                       "20122013"=>3.86,
                       "20162017"=>4.75,
                       "20142015"=>3.75,
                       "20152016"=>3.88,
                       "20132014"=>4.33
                     }
-    assert_equal expected_hash, @stat_tracker.average_goals_by_season
+    assert_equal expected_hash, @stat_tracker.average_goals_per_season
```

Comment:
The method name that Iteration 2 calls for is `average_goals_by_season` - if you want to discard/revert the changes on this file, my "quick fix" pull request updates it to the way the spec harness expects.